shared_resources/python-modules/python/shared/dynamodb/jobs.py

```
@@ -23,6 +24,59 @@ def query_clinic_job(job_id):
     return response.get("Item")
 
 
+def scan_pending_jobs():
+    # Get datetime 2 days ago
+    threshold = datetime.now(timezone.utc) - timedelta(days=2)
+
+    # Filter pending jobs from DynamoDB
+    response = dynamodb_client.scan(
+        TableName=DYNAMO_CLINIC_JOBS_TABLE,
+        FilterExpression="job_status = :status",
+        ExpressionAttributeValues={":status": {"S": "pending"}},
+    )
+
+    items = response.get("Items", [])
+    old_pending_jobs = []
+
+    for item in items:
+        created_date_str = item.get("created_at", {}).get("S")
+        if not created_date_str:
+            continue
+
+        try:
+            created_dt = datetime.fromisoformat(created_date_str)
+        except ValueError:
+            continue
```

This exception may have been hiding an error with your `create_date_str` format. Make sure that items are deleted when expected in testing.

---

i'll update using threshold time